transient/sshfs.py

```diff
@@ -21,6 +26,11 @@
 
 _PROVISION_SCRIPTS = {"RHEL": _RHEL_PROVISION_SCRIPT, "FEDORA": _RHEL_PROVISION_SCRIPT}
 
+global provision_lock
+provision_lock = threading.Lock()
+global provision_success
+provision_success = None
```

Comment:
This should be type annotated. In general I'm ok with things like `provision_lock = threading.Lock()` not being annotated, as the type is right there, but for things like `None` we know the value is definitely not always going to be `None`, so knowing the actual type is useful.

Reply:
Also, this and the above provision_lock seems like they should be class variables on `SshfsThread` given that the semaphore is there already anyway

Reply:
Done and done. Luckily, this also allows us to avoid mypy's insistence on `global` keywords here.